### PR TITLE
583 Absence/tardy counter

### PR DESCRIPTION
**Pull request by iamsitting**
Displays count of (excused/unexcused) absences and tardies. Issue #583
I can't get the excused count to show up though.

### CODE REVIEW

**Comment by iamsitting** [comment]
Summary.js
```javascript
let unexcused_absences=p.eventsRolls.filter(esr => esr.event.status.roll==='absent'&&esr.event.status.slip!=='approved')
let unexcused_tardies=p.eventsRolls.filter(esr => esr.event.status.roll==='tardy'&&esr.event.status.slip!=='approved')
let excused_absences=p.eventsRolls.filter(esr => esr.event.status.roll==='absent'&&esr.event.status.slip==='approved')
let excused_tardies=p.eventsRolls.filter(esr => esr.event.status.roll==='tardy'&&esr.event.status.slip==='approved')
```
The logic above doesn't work because:
(1) if status.roll is 'tardy' or 'absent', status.slip does not exist
(2) if status.slip is 'approved', status.roll is 'excused'

**Comment by billyeh** [comment]
@joeyu23

**Comment by joeyu23** [comment]
Fixed logic. @iamsitting @billyeh. Please review at earliest convenience.

**Comment by iamsitting** [comment]
@joeyu23 Works mostly well. There needs to be a condition to handle uniform tardies. The un/excused tardies should include rolls marked as 'uniform' or 'U' (or something like that).

File: ap/leaveslips/models.py [hunk 1]
```diff
@@ -74,6 +73,9 @@ class LeaveSlip(models.Model):
 
   informed = models.BooleanField(blank=True, default=False, verbose_name='informed TA')  # informed TA
 
+  def get_trainee_requester(self):
```
Comment (billyeh) [inline ap/leaveslips/models.py]: This is duplicated on line 99. Maybe got merged in.

**Comment by joeyu23** [comment]
@iamsitting Addressed the issue in the latest commit

**Comment by iamsitting** [comment]
Addressed duplicate code. Everything else LGTM.

**Pull request merged**